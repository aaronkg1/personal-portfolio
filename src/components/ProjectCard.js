import React from "react";

const ProjectCard = (props) => {
  const { title, description, url } = props;

  return (
    <li key={`${title}ID`}>
      <h2></h2>
    </li>
  );
};
