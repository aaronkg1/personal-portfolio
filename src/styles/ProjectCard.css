.project-card {
  max-height: 350px;
  list-style: none;
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 100px;
  font-family: "Nunito", sans-serif;
  color: white;
  background-color: transparent;
  border-radius: 10px;
}
.project-card:nth-child(even) {
  flex-direction: row-reverse;
}

.info-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 30px;
  font-size: 1.5em;
}

.button-wrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 10px;
}

.hover-zoom {
  height: 300px;
}

.hover-zoom:hover {
  transform: scale(1.2);
}

.project-button {
  outline: none;
  border: none;
  padding: 10px;
  font-size: 0.7em;
  border-radius: 15px;
  border: solid rgb(225, 102, 20) 3px;
  background-color: transparent;
  cursor: pointer;
  color: white;
  transition: border-color 0.8s;
  display: flex;
  gap: 10px;
  font-size: 1.2em;
}

.project-button a {
  text-decoration: none;
  color: white;
}

.light .project-button {
  border-color: rgb(131, 0, 164);
}

@media screen and (max-width: 1200px) {
  .project-card {
    gap: 30px;
    max-width: 100vw;
    flex-direction: column;
  }

  .project-card:nth-child(even) {
    flex-direction: column;
  }

  .hover-zoom {
    max-height: 150px;
  }
}
