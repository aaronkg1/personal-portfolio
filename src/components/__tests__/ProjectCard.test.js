import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

import ProjectCard from "../ProjectCard";

afterEach(() => {
  cleanup();
});

test("should render ProjectCard component", () => {
  render(
    <ProjectCard
      project={{
        title: "Test Title",
        description: "Test Description",
        url: "https://testurl.test",
        liveLink: "",
        repository: "",
      }}
    />
  );
  const ProjectCardElement = screen.getByTestId("ProjectCard-1");
  const cardImg = screen.getByTestId("card-image");
  expect(ProjectCardElement).toBeInTheDocument();
  expect(ProjectCardElement).toHaveTextContent("Test Title");
  expect(ProjectCardElement).toHaveTextContent("Test Description");
  expect(ProjectCardElement).toContainElement(cardImg);
});

test("should render multiple todos", () => {
  const projects = [
    {
      title: "Test One",
      description: "Test One Is Better",
      url: "https://testOne.faketest",
      liveLink: "",
      repository: "",
    },
    {
      title: "Test Two",
      description: "Test Two Is Better",
      url: "https://testTwo.faketest",
      liveLink: "",
      repository: "",
    },
  ];

  render(<ProjectCard project={projects[0]} />);
  const ProjectCardElement = screen.getByTestId("ProjectCard-1");
  expect(ProjectCardElement).toContainHTML("<li");
});

test("matches snapshot", () => {
  const project = {
    title: "SnapShot Test",
    description: "Snapshot updates",
    url: "https://testOne.faketest",
    liveLink: "",
    repository: "",
  };
  const tree = renderer.create(<ProjectCard project={project} />).toJSON();
  expect(tree).toMatchSnapshot();
});
