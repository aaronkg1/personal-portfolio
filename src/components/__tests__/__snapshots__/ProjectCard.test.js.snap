// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<li
  className="project-card"
  data-testid="ProjectCard-1"
>
  <div
    className="image-wrapper"
  >
    <img
      alt="SnapShot Test"
      className="hover-zoom"
      data-testid="card-image"
      src="https://testOne.faketest"
    />
  </div>
  <div
    className="info-container"
  >
    <h2>
      SnapShot Test
    </h2>
    <p>
      Snapshot updates
    </p>
    <div
      className="button-wrapper"
    >
      <button
        className="project-button"
      >
        <a
          alt="SnapShot Test"
          href=""
        >
          Live App
        </a>
      </button>
      <button
        className="project-button"
      >
        <a
          alt="SnapShot Test repository"
          href=""
        >
          Repository
        </a>
      </button>
    </div>
  </div>
</li>
`;
